test/provisioner/provision_azure.go

```diff
@@ -8,37 +8,355 @@ package provisioner
+
+	return nil
+}
+
+func createVnetSubnet() error {
+	addressPrefix := "10.2.0.0/16"
+	subnetAddressPrefix := "10.2.0.0/24"
+	vnetParams := network.VirtualNetwork{
+		Location: &AzureProps.Location,
+		Name:     &AzureProps.VnetName,
+		VirtualNetworkPropertiesFormat: &network.VirtualNetworkPropertiesFormat{
+			AddressSpace: &network.AddressSpace{
+				AddressPrefixes: &[]string{addressPrefix},
+			},
+			Subnets: &[]network.Subnet{
+				{
+					Name: &AzureProps.SubnetName,
+					SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
+						AddressPrefix: &subnetAddressPrefix,
+					},
+				},
+			},
+		},
+	}
+
+	// Create the virtual network
+	log.Infof("Creating  vnet %s in resource group %s with addressPrefix %s subnetAddressPrefix %s.\n", AzureProps.VnetName, AzureProps.ResourceGroupName, addressPrefix, subnetAddressPrefix)
+	_, err := AzureProps.ManagedvnetClient.CreateOrUpdate(context.Background(), AzureProps.ResourceGroupName, AzureProps.VnetName, vnetParams)
+	if err != nil {
+		return err
+	}
+
+	subnet, err := AzureProps.ManagedsubnetClient.Get(context.Background(), AzureProps.ResourceGroupName, AzureProps.VnetName, AzureProps.SubnetName, "")
+	if err != nil {
+		return err
+	}
+
+	AzureProps.SubnetID = *subnet.ID
+
+	if len(AzureProps.SubnetID) <= 0 {
+		return errors.New("SubnetID is empty, unknown error happened when create Subnet.")
+	}
+
+	log.Infof("Successfully Created vnet %s in resource group %s.\n", AzureProps.VnetName, AzureProps.ResourceGroupName)
+
+	return nil
+}
+
+func deleteVNet() error {
+	log.Infof("Deleting vnet %s in resource group %s.\n", AzureProps.VnetName, AzureProps.ResourceGroupName)
+	_, err := AzureProps.ManagedvnetClient.Delete(context.Background(), AzureProps.ResourceGroupName, AzureProps.VnetName)
+	if err != nil {
+		if isNotFoundError(err) {
+			return nil
+		}
+		fmt.Println("Failed to delete the virtual network: ", err)
+		return err
+	}
+
+	log.Infof("Successfully Deleted vnet %s in resource group %s.\n", AzureProps.VnetName, AzureProps.ResourceGroupName)
+	return nil
+}
+
+func createResourceImpl() error {
+	err := createResourceGroup()
+	if err != nil {
+		return err
+	}
+
+	log.Info("waiting for the Resource group to be available before creating vnet...")
+	time.Sleep(60 * time.Second)
+	return createVnetSubnet()
+}
+
+func deleteResourceImpl() error {
+	err := deleteVNet()
+	if err != nil {
+		return err
+	}
+	return deleteResourceGroup()
+}
+
+func isAzureClusterReady(resourceGroupName string, clusterName string) (bool, error) {
+	cluster, err := AzureProps.ManagedaksClient.Get(context.Background(), resourceGroupName, clusterName)
+	if err != nil {
+		return false, fmt.Errorf("failed to get cluster: %v", err)
+	}
+
+	if *cluster.ProvisioningState != "Succeeded" {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func isNotFoundError(err error) bool {
+	if typedError, ok := err.(autorest.DetailedError); ok {
+		if typedError.StatusCode == 404 {
```

Comment:
I think we can use the http status code `StatusNotFound` from here: https://pkg.go.dev/net/http